apps/voucher/app/create/page.tsx

```diff
@@ -0,0 +1,126 @@
+"use client"
+import { useState } from "react"
+
+import CreatePageAmount from "@/components/Create/CreatePageAmount/CreatePageAmount"
+import CreatePagePercentage from "@/components/Create/CreatePagePercentage/CreatePagePercentage"
+import { Currency, Wallet } from "@/lib/graphql/generated"
+import useSatsPrice from "@/hooks/useSatsPrice"
+import PageLoadingComponent from "@/components/Loading/PageLoadingComponent"
+import { calculateCommission, getWalletDetails } from "@/utils/helpers"
+import ConfirmModal from "@/components/Create/ConifrmModal/ConfirmModal"
+import InfoComponent from "@/components/InfoComponent/InfoComponent"
+import useRealtimePrice from "@/hooks/useRealTimePrice"
+import { DEFAULT_CURRENCY } from "@/config/appConfig"
+import { gql } from "@apollo/client"
+import { useSession } from "next-auth/react"
+
+gql`
+  mutation CreateWithdrawLink($input: CreateWithdrawLinkInput!) {
+    createWithdrawLink(input: $input) {
+      commissionPercentage
+      createdAt
+      id
+      identifierCode
+      salesAmount
+      status
+      uniqueHash
+      userId
+      voucherAmount
+      voucherSecret
+      paidAt
+    }
+  }
+`
+
+export default function CreatePage() {
+  const session = useSession()
+
+  const { usdToSats } = useSatsPrice()
+  const storedCurrency =
+    typeof window !== "undefined" ? localStorage.getItem("currency") : null
+  const storedCommission =
+    typeof window !== "undefined" ? localStorage.getItem("commission") : null
+
+  const [currency, setCurrency] = useState<Currency>(
+    storedCurrency ? JSON.parse(storedCurrency) : DEFAULT_CURRENCY,
+  )
+  const [commissionPercentage, setCommissionPercentage] = useState<string>(
+    storedCommission || "0",
+  )
+  const { currencyToCents, hasLoaded } = useRealtimePrice(currency.id)
+  const [amount, setAmount] = useState<string>("0")
```

Comment:
string? shouldn't it be a number?

Reply:
Yes, but we have some numpad validations that require it to be in a string, but we can do refactoring there to make int number.